scripts/globals/npc_util.lua

```diff
@@ -7,25 +7,34 @@ npcUtil = {};
 -- Picks a new position for an NPC and excluding the current position.
 -- INPUT: npc = npcID, position = 2D table with coords: index, {x, y, z}
 -- RETURN: table index
-function npcUtil.pickNewPosition(npc, positionTable)
-
+function npcUtil.pickNewPosition(npc, positionTable, allowCurrentPosition)
     local npc = GetNPCByID(npc)
     local positionIndex = 1 -- Default to position one in the table if it can't be found.
     local tableSize = 0
+    local allowCurrentPosition = allowCurrentPosition or false;
+
     for i, v in ipairs(positionTable) do   -- Looking for the current position
 
-        -- Finding by comparing the NPC's coords
-        if (math.floor(v[1]) == math.floor(npc:getXPos()) and math.floor(v[2]) == math.floor(npc:getYPos()) and math.floor(v[3]) == math.floor(npc:getZPos())) then
-            positionIndex = i; -- Found where the NPC is!
+        if (allowCurrentPosition == "false") then
```

Comment:
I tried in many different ways, comparing as a string was the only way it seemd to work. I will give this a try, maybe I just didn't see the obvious